CMakeLists.txt

```diff
@@ -43,22 +43,73 @@ endif()
 # Project structure ###########################################################
 #
 # temporary build directories
-if(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
-    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+if(NOT openPMD_ARCHIVE_OUTPUT_DIRECTORY)
+    if(CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+        set(openPMD_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY})
+    else()
+        set(openPMD_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+    endif()
 endif()
-if(NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
-    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+if(NOT openPMD_LIBRARY_OUTPUT_DIRECTORY)
+    if(CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+        set(openPMD_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+    else()
+        set(openPMD_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+    endif()
 endif()
-if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
-    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+if(NOT openPMD_RUNTIME_OUTPUT_DIRECTORY)
+    if(CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+        set(openPMD_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+    else()
+        set(openPMD_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+    endif()
 endif()
```

Comment:
FYI:
there is also
* [CMAKE_PDB_OUTPUT_DIRECTORY](https://cmake.org/cmake/help/latest/variable/CMAKE_PDB_OUTPUT_DIRECTORY.html#variable:CMAKE_PDB_OUTPUT_DIRECTORY)
* [CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY.html#variable:CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY)